src/process-cache.js

```diff
@@ -1,129 +1,61 @@
-             *
-             * We don't care about ignore files, because we're giving a list of specific
-             * files, but we do care about:
-             *  - comment matchers
-             *  - the root marker
-             */
-            const updateCommandParts = [];
-            updateCommandParts.push(getLaunchString());
-            const commentsArg = options.comments.sort().join(",");
-            if (commentsArg !== defaultArgs.comments) {
-                updateCommandParts.push("-c");
-                updateCommandParts.push(`"${commentsArg}"`);
-            }
-            const rootMarker = options.rootMarker;
-            if (rootMarker != null) {
-                updateCommandParts.push("-m");
-                updateCommandParts.push(`"${rootMarker}"`);
+        return async (options, file, cache) => {
+            if (!(await fileValidator(options, file, cache, log))) {
+                violationFileNames.push(file);
             }
-            updateCommandParts.push("-u");
-            updateCommandParts.push(
-                violationFileNames.map(cwdRelativePath).join(" "),
-            );
-
-            log.log(updateCommandParts.join(" "));
         };
-
-        if (autoFix) {
-            if (options.dryRun) {
-                log.group(
-                    `${violationFileNames.length} file(s) would have been fixed. To fix, run:`,
-                );
-                outputRerunCommand();
-                log.groupEnd();
-            } else {
-                // Output a summary of what we fixed.
-                log.info(`Fixed ${violationFileNames.length} file(s)`);
-            }
-        } else {
-            // Output how to fix any violations we found if we're not running
-            // autofix.
-            log.log("");
-            const errorMsg = log.errorsLogged
-                ? "🛑  Desynchronized blocks detected and parsing errors found. Fix the errors, update the blocks, then update the sync-start tags using:"
-                : "🛠  Desynchronized blocks detected. Check them and update as required before resynchronizing:";
-            log.group(`${errorMsg}`);
-            outputRerunCommand();
-            log.groupEnd();
-            return ErrorCodes.DESYNCHRONIZED_BLOCKS;
+    };
+
+    const validator = getValidator();
+
+    for (const file of Object.keys(cache)) {
+        try {
+            await validator(options, file, cache);
+        } catch (e) {
+            log.error(
+                `${Format.cwdFilePath(
+                    cwdRelativePath(file),
+                )} update encountered error: ${e.message}`,
+            );
         }
     }
 
-    if (log.errorsLogged) {
-        log.log("🛑  Errors occurred during processing");
-        return ErrorCodes.PARSE_ERRORS;
+    if (json) {
+        return outputJson(log, jsonItems);
     }
 
-    log.log("🎉  Everything is in sync!");
-    return ErrorCodes.SUCCESS;
+    return outputText(options, violationFileNames, log);
```

Comment:
What do you think about having a `getOutput` function like `getValidator` that returns a function with a standard set of args, like the validator. Then the output function itself could decide what to do with those args.

Long term, we might only use the json validator or the fix validator or a combination, and then the output functions would do all the work based off that, so keeping a single path and clean interface through will help with that later refactoring - keeping the branching and one-off differences localized to the functions themselves.

Reply:
Will do.